### PR TITLE
Drop Python 2 support 

### PR DESCRIPTION
**Pull request by duckduckgrayduck**

### CODE REVIEW

**Comment by duckduckgrayduck** [comment]
Will close https://github.com/MuckRock/python-documentcloud/issues/10 and get us one step closer to running black & pylint on commits and PRs

**Comment by duckduckgrayduck** [comment]
This is where we are at with Pylint: 

************* Module documentcloud.addon
documentcloud/addon.py:27:8: E1111: Assigning result of a function call, where the function has no return (assignment-from-no-return)
documentcloud/addon.py:27:8: W0612: Unused variable 'client' (unused-variable)
documentcloud/addon.py:68:34: C0209: Formatting a regular string which could be an f-string (consider-using-f-string)
documentcloud/addon.py:122:17: W1514: Using open without explicitly specifying an encoding (unspecified-encoding)
documentcloud/addon.py:178:19: W3101: Missing timeout argument for method 'requests.put' can cause your program to hang indefinitely (missing-timeout)
documentcloud/addon.py:202:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
************* Module documentcloud.documents
documentcloud/documents.py:190:0: C0301: Line too long (93/88) (line-too-long)
documentcloud/documents.py:378:0: C0301: Line too long (89/88) (line-too-long)
documentcloud/documents.py:65:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/documents.py:280:18: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
documentcloud/documents.py:372:4: R0914: Too many local variables (18/15) (too-many-locals)
documentcloud/documents.py:396:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
documentcloud/documents.py:408:12: W1201: Use lazy % formatting in logging functions (logging-not-lazy)
documentcloud/documents.py:432:20: W1201: Use lazy % formatting in logging functions (logging-not-lazy)
documentcloud/documents.py:445:16: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
documentcloud/documents.py:448:34: R1732: Consider using 'with' for resource-allocating operations (consider-using-with)
documentcloud/documents.py:453:24: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
documentcloud/documents.py:467:20: W1201: Use lazy % formatting in logging functions (logging-not-lazy)
documentcloud/documents.py:372:4: R0912: Too many branches (14/12) (too-many-branches)
documentcloud/documents.py:492:24: W1202: Use lazy % formatting in logging functions (logging-format-interpolation)
documentcloud/documents.py:512:20: W1201: Use lazy % formatting in logging functions (logging-not-lazy)
************* Module documentcloud.exceptions
documentcloud/exceptions.py:19:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
************* Module documentcloud.base
documentcloud/base.py:102:12: C0209: Formatting a regular string which could be an f-string (consider-using-f-string)
documentcloud/base.py:123:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/base.py:172:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/base.py:194:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/base.py:203:12: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/base.py:216:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
************* Module documentcloud.client
documentcloud/client.py:129:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
documentcloud/client.py:145:8: W1203: Use lazy % formatting in logging functions (logging-fstring-interpolation)
documentcloud/client.py:172:16: W0707: Consider explicitly re-raising using 'raise DoesNotExistError(response=exc.response) from exc' (raise-missing-from)
documentcloud/client.py:174:16: W0707: Consider explicitly re-raising using 'raise APIError(response=exc.response) from exc' (raise-missing-from)
************* Module documentcloud.toolbox
documentcloud/toolbox.py:9:0: C0411: standard import "from urllib.parse import urlparse" should be placed before "import requests" (wrong-import-order)
documentcloud/toolbox.py:10:0: C0411: standard import "from itertools import zip_longest" should be placed before "import requests" (wrong-import-order)
************* Module documentcloud.projects
documentcloud/projects.py:17:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/projects.py:26:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
documentcloud/projects.py:102:4: W0237: Parameter 'id_' has been renamed to 'id' in overriding 'ProjectClient.get' method (arguments-renamed)
documentcloud/projects.py:102:4: W0237: Parameter 'expand' has been renamed to 'title' in overriding 'ProjectClient.get' method (arguments-renamed)
documentcloud/projects.py:118:15: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)

------------------------------------------------------------------
Your code has been rated at 9.52/10

File: documentcloud/base.py [hunk 1]
```diff
@@ -39,10 +33,10 @@ def __init__(
         ]
 
     def __repr__(self):
-        return "<APIResults: {!r}".format(self.results)  # pragma: no cover
```
Comment (mitchelljkotler) [inline documentcloud/base.py:42]: You lost the !r specifier here

https://realpython.com/python-f-strings/#using-an-objects-string-representations-in-f-strings

File: documentcloud/client.py [hunk 2]
```diff
@@ -131,12 +126,12 @@ def user_id(self):
     def _request(self, method, url, raise_error=True, **kwargs):
         """Generic method to make API requests"""
         # pylint: disable=method-hidden
-        logger.info("request: %s - %s - %s", method, url, kwargs)
```
Comment (mitchelljkotler) [inline documentcloud/client.py]: Logs should be lazy, as pylint is now warning:

https://medium.com/flowe-ita/logging-should-be-lazy-bc6ac9816906

**Review by mitchelljkotler — CHANGES_REQUESTED** [review]

**Pull request merged**